src/operation/get-parameters-as-json-schema.ts

```diff
@@ -122,32 +116,33 @@ export default function getParametersAsJsonSchema(
   /**
    *
    */
-  function getRequestBody(): SchemaWrapper {
-    const schema = getSchema(operation, api);
-    if (!schema || !schema.schema) return null;
+  function transformRequestBody(): SchemaWrapper {
+    const requestBody = operation.getRequestBody();
+    if (!requestBody || !Array.isArray(requestBody)) return null;
 
-    const type = schema.type === 'application/x-www-form-urlencoded' ? 'formData' : 'body';
-    const requestBody = schema.schema;
+    const [mediaType, mediaTypeObject] = requestBody;
+    const type = mediaType === 'application/x-www-form-urlencoded' ? 'formData' : 'body';
 
     // If this schema is completely empty, don't bother processing it.
-    if (Object.keys(requestBody.schema).length === 0) {
+    if (!Object.keys(mediaTypeObject.schema).length) {
       return null;
     }
 
-    const examples = [];
-    if ('example' in requestBody) {
-      examples.push({ example: requestBody.example });
-    } else if ('examples' in requestBody) {
-      examples.push({ examples: requestBody.examples });
+    // @todo this `examples` array be `Array<RMOAS.SchemaObject>` but that doesn't like the `examples` schema
+    const examples = [] as Array<any>;
```

Comment:
This `Array<any>` is annoying but I had a lot of trouble with this block and the `prevSchemas` type in `toJSONSchema`.

For example if this is just `const examples = [];` TS throws this:

```
src/operation/get-parameters-as-json-schema.ts:142:73 - error TS2322: Type '({ example: any; } | { examples: Record<string, ReferenceObject | ExampleObject> | { [media: string]: ReferenceObject | ExampleObject; }; })[]' is not assignable to type 'PrevSchemasType'.
  Type '{ example: any; } | { examples: Record<string, ReferenceObject | ExampleObject> | { [media: string]: ReferenceObject | ExampleObject; }; }' is not assignable to type 'SchemaObject'.
    Type '{ examples: Record<string, OpenAPIV3_1.ReferenceObject | OpenAPIV3.ExampleObject> | { [media: string]: OpenAPIV3.ReferenceObject | OpenAPIV3.ExampleObject; }; }' is not assignable to type 'SchemaObject'.
      Type '{ examples: Record<string, OpenAPIV3_1.ReferenceObject | OpenAPIV3.ExampleObject> | { [media: string]: OpenAPIV3.ReferenceObject | OpenAPIV3.ExampleObject; }; }' is not assignable to type 'JSONSchema7 & { deprecated?: boolean; readOnly?: boolean; writeOnly?: boolean; example?: unknown; examples?: unknown[]; nullable?: boolean; xml?: unknown; externalDocs?: unknown; components?: ComponentsObject; 'x-readme-ref-name'?: string; }'.
        Type '{ examples: Record<string, OpenAPIV3_1.ReferenceObject | OpenAPIV3.ExampleObject> | { [media: string]: OpenAPIV3.ReferenceObject | OpenAPIV3.ExampleObject; }; }' is not assignable to type 'JSONSchema7'.
          Types of property 'examples' are incompatible.
            Type 'Record<string, ReferenceObject | ExampleObject> | { [media: string]: ReferenceObject | ExampleObject; }' is not assignable to type 'JSONSchema7Type'.
              Type 'Record<string, ReferenceObject | ExampleObject>' is not assignable to type 'JSONSchema7Type'.
                Type 'Record<string, ReferenceObject | ExampleObject>' is not assignable to type 'string'.

142     const cleanedSchema = toJSONSchema(requestSchema, { globalDefaults, prevSchemas: examples, refLogger });
                                                                            ~~~~~~~~~~~

  src/lib/openapi-to-json-schema.ts:33:3
    33   prevSchemas?: PrevSchemasType;
         ~~~~~~~~~~~
    The expected type comes from property 'prevSchemas' which is declared here on type 'toJsonSchemaOptions'


Found 1 error.
```

If it's `const examples = [] as Array<RMOAS.SchemaObject>;` TS throws this:

```
src/operation/get-parameters-as-json-schema.ts:136:23 - error TS2322: Type 'Record<string, ReferenceObject | ExampleObject> | { [media: string]: ReferenceObject | ExampleObject; }' is not assignable to type 'unknown[] | (any[] & unknown[]) | (JSONSchema6Type[] & unknown[]) | (JSONSchema7Type & unknown[])'.
  Type 'Record<string, ReferenceObject | ExampleObject>' is not assignable to type 'unknown[] | (any[] & unknown[]) | (JSONSchema6Type[] & unknown[]) | (JSONSchema7Type & unknown[])'.
    Type 'Record<string, ReferenceObject | ExampleObject>' is not assignable to type 'string & unknown[]'.
      Type 'Record<string, ReferenceObject | ExampleObject>' is not assignable to type 'string'.

136       examples.push({ examples: mediaTypeObject.examples });
                          ~~~~~~~~

  node_modules/openapi-types/dist/index.d.ts:80:9
    80         examples?: OpenAPIV3.BaseSchemaObject['example'][];
               ~~~~~~~~
    The expected type comes from property 'examples' which is declared here on type 'SchemaObject'


Found 1 error.
```

I get it's a problem with the `ReferenceObject`, as usual, I'm just not sure how to fight that battle here.

Reply:
I dug into this a bit, I don't think it's a ReferenceObject issue. I think it's a legit type inconsistency, and either a bug or something we're magically supporting.

If we go with the second option, `Array<RMOAS.SchemaObject>` it expects the examples to be `Array<any>`, because that's the spec for JSON schema.

But `mediaTypeObject.exexamples` isn't `Array<any>` it's `Record<string, ReferenceObject | ExampleObject>` as defined in OAS.

It seems line 136 of get-parameters-as-json-schema is mixing up JSON Schema examples and OAS examples.

Line 134 works because it's actually grabbing the value via `example`, but 136 needs to grab the `value` prop off each example.

e.g.

```
    // @todo this `examples` array be `Array<RMOAS.SchemaObject>` but that doesn't like the `examples` schema
    const examples = [] as Array<RMOAS.SchemaObject>;
    if ('example' in mediaTypeObject) {
      examples.push({ example: mediaTypeObject.example });
    } else if ('examples' in mediaTypeObject) {
      examples.push({
        examples: Object.values(mediaTypeObject.examples).map((example: RMOAS.ExampleObject) => example.value),
      });
    }
```